### PR TITLE
[SystemInfo] add system_info_cellular_network_ivi.cc to fit ivi system

### PR DESCRIPTION
**Pull request by liyingzh**
Follow  system_info_sim_ivi.cc(#369), add system_info_cellular_network_ivi.cc to fit ivi system. Also fix the lac and cellId attribute error value.

BUG=XWALK-2362

### CODE REVIEW

File: src/system_info/system_info_cellular_network.cc [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
```
Comment (halton) [inline src/system_info/system_info_cellular_network.cc]: nit, 2015

File: src/system_info/system_info_cellular_network_ivi.cc [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+               g_strcmp0(val, "denied") == 0) {
+      status_ = "ON";
+    } else if (g_strcmp0(val, "roaming") == 0) {
+      status_ = "ON";
+      roaming_ = true;
+    } else {
+      status_ = "Unknown";
+    }
+  } else if (g_strcmp0(key, "RoamingAllowed") == 0) {
+    roaming_allowed_ = g_variant_get_boolean(var_val) ? true : false;
```
Comment (halton) [inline src/system_info/system_info_cellular_network_ivi.cc]: `roaming_allowed_ = g_variant_get_boolean(var_val)` is fine.

File: src/system_info/system_info_cellular_network_ivi.cc [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      apn_(""),
+      ipAddress_(""),
+      ipv6Address_(""),
+      mcc_(0),
+      mnc_(0),
+      cellId_(0),
+      lac_(0),
+      isRoaming_(false),
+      // FIXME(liying): NOT SUPPORTED
+      isFlightMode_(false),
```
Comment (liyingzh) [inline src/system_info/system_info_cellular_network_ivi.cc]: isFlightMode atrribute can not be got from ofono, and acording to [Tizen api spec] (https://developer.tizen.org/dev-guide/2.3.0/org.tizen.mobile.native.apireference/group__CAPI__SYSTEM__SYSTEM__SETTINGS__MODULE.html#ga56c8fa435516884c5648efecdd871eaa), there will be SYSTEM_SETTINGS_KEY_NETWORK_FLIGHT_MODE in ````#include <system_settings.h>````. So will support isFlightMode in the future as system_settings package does not have the feature now.

File: src/system_info/system_info_sim_ivi.cc [hunk 4]
```diff
@@ -23,11 +14,18 @@ SysInfoSim::SysInfoSim()
       msin_(""),
       spn_(""),
       conn_(NULL) {
-  InitDbusConnection();
+  conn_ = system_info::GetDbusConnection();
+  if (!conn_)
+    return;
+  modem_path_ = system_info::OfonoGetModemPath(conn_);
```
Comment (halton) [inline src/system_info/system_info_sim_ivi.cc]: `system_info::OfonoGetModemPath()` seems will be called in multiple places. To saving the the dbus calls, I'd suggest to rewrite it to return a static result if it is ever be called. @zolkis any comment?

File: src/system_info/system_info_cellular_network_ivi.cc (hunk 3 above)
Comment (halton) [inline src/system_info/system_info_cellular_network_ivi.cc]: Thanks for the information, better to have that information as comment.

**Comment by liyingzh** [comment]
isFlightMode atrribute can not be got from ofono, and acording to [Tizen api spec] (https://developer.tizen.org/dev-guide/2.3.0/org.tizen.mobile.native.apireference/group__CAPI__SYSTEM__SYSTEM__SETTINGS__MODULE.html#ga56c8fa435516884c5648efecdd871eaa), there will be SYSTEM_SETTINGS_KEY_NETWORK_FLIGHT_MODE in ````#include <system_settings.h>````. So will support isFlightMode in the future as system_settings package does not have the feature now, also VCONFKEY_TELEPHONY_FLIGHT_MODE is not available in IVI now.

File: src/system_info/system_info_sim_ivi.cc (hunk 4 above)
Comment (zolkis) [inline src/system_info/system_info_sim_ivi.cc]: No, the whole point is to check run-time. Modems can be powered up and down and enabled/disabled. Saving the dbus call means implementing state management here, which is more complex than a sparse call.

However, in the system_info::OfonoGetModemPath() method, it doesn't seem correct to return NULL (should return std::string value).

File: src/system_info/system_info_cellular_network_mobile.cc [hunk 5]
```diff
@@ -8,7 +8,20 @@
 #include <net_connection.h>
 #include <system_info.h>
 
-const std::string SysInfoCellularNetwork::name_ = "CELLULAR_NETWORK";
+SysInfoCellularNetwork::SysInfoCellularNetwork()
+    : status_(""),
+      apn_(""),
+      ipAddress_(""),
+      ipv6Address_("NOT SUPPORTTED"),
```
Comment (deqing) [inline src/system_info/system_info_cellular_network_mobile.cc]: nit, `NOT SUPPORTED`

File: src/system_info/system_info_sim_ivi.cc (hunk 4 above)
Comment (halton) [inline src/system_info/system_info_sim_ivi.cc]: forget my comment then. 
@liyingzh , please fix the `system_info::OfonoGetModemPath()` issue that @zolkis mentioned.

**Comment by halton** [comment]
LGTM, will merge after trybot get green.

**Pull request merged**